config/routes.rb

```diff
@@ -84,8 +116,12 @@ def matches?(_)
   get '(:lang)/listings/for-rent' => 'directory#rent', lang: /(en|es|zh|tl)/
   get '(:lang)/listings/for-sale' => 'directory#sale', lang: /(en|es|zh|tl)/
   # TODO: Paths on Bloom needs to be configurable
-  get '(:lang)/listings/:id' => 'listing#index', lang: /(en|es|zh|tl)/, as: 'listing'
-  get '(:lang)/listings/:id/how-to-apply' => 'listing#how_to_apply', lang: /(en|es|zh|tl)/
+  get '(:lang)/listings/:id' => 'listing#index', lang: /(en|es|zh|tl)/
+
+  # If the constraint doesn't pass, it falls back to the next match
+  get '(:lang)/listings/:id/how-to-apply' => 'listing#how_to_apply', lang: /(en|es|zh|tl)/, constraints: HowToApplyConstraint.new
+  get '(:lang)/listings/:id/how-to-apply', to: redirect('%{lang}/listings/%{id}')
```

Comment:
issue: This part of the redirect url is causing 500 errors: `%{lang}`. Not sure how to fix it thought.

Reply:
found a fix:
```rb
  scope '(:lang)', lang: /(en|es|zh|tl)/ do
    get '/listings/:id/how-to-apply', to: redirect('%{lang}/listings/%{id}')
  end
```

Reply:
hm, I see the error you're talking about, but the fix you provided isn't working for me

Reply:
I pushed a different fix, I think - lmk what you think!

Reply:
looks good, I've realized that the issue was related to handling both the non english url prefix and the no-url-prefix situation